### ISSUE TITLE
Figure out why federation does not work with 1.3.0 kubectl

### ISSUE THREAD

**Issue by nikhiljindal** [kubernetes/kubernetes#29469]
From slack channel:

> @sethkarlo: I’ve successfully built and deployed the federation api, but kubectl is not happy talking with it. Using curl alone I can browse the api, but when I use kubectl I get an odd error, I’m assuming this is an issue with kubectl rather than federation, but has anyone else seen this/can recommend a fix?
> ```
> F0722 13:43:22.090808   46152 helpers.go:108] error: group map[:0xc8200e7f10 apps:0xc8200e7f80 authorization.k8s.io:0xc820480000 extensions:0xc82004c070 rbac.authorization.k8s.io:0xc8200e6070 authentication.k8s.io:0xc8200e6230 federation:0xc8200e7ea0 autoscaling:0xc820480070 batch:0xc8204800e0 componentconfig:0xc820480150 policy:0xc8200e6000] is already registered
> ```
> @sethkarlo: Figured it out. In case it hits anyone else, building from release 1.3.0 will not work with federation, you explicitly needed kubectl version 1.3.0-beta1
> 

cc @kubernetes/sig-cluster-federation @jianhuiz

**Comment by madhusudancs** [comment]
Any idea if this is still an issue with v1.3.3?

**Comment by Seth-Karlo** [comment]
Hey guys, some more info for you:

```$ kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.0", GitCommit:"283137936a498aed572ee22af6774b6fb6e9fd94", GitTreeState:"clean", BuildDate:"2016-07-01T19:26:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.0-beta.1", GitCommit:"1c7855093eb999c8f3cedb19d3467cd7b691e35e", GitTreeState:"clean", BuildDate:"2016-06-17T00:53:24Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
$ kubectl get cs
error: group map[federation:0xc8203daf50 authorization.k8s.io:0xc8203db0a0 batch:0xc8203db180 componentconfig:0xc8203db1f0 rbac.authorization.k8s.io:0xc8203db340 authentication.k8s.io:0xc8203db420 :0xc8203dafc0 apps:0xc8203db030 autoscaling:0xc8203db110 extensions:0xc8203db260 policy:0xc8203db2d0] is already registered```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.0-beta.1", GitCommit:"1c7855093eb999c8f3cedb19d3467cd7b691e35e", GitTreeState:"clean", BuildDate:"2016-06-17T00:58:51Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.0-beta.1", GitCommit:"1c7855093eb999c8f3cedb19d3467cd7b691e35e", GitTreeState:"clean", BuildDate:"2016-06-17T00:53:24Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
$ kubectl get cs
Unable to list "componentstatuses": the server could not find the requested resource```

```$ ./kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:29:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.0-beta.1", GitCommit:"1c7855093eb999c8f3cedb19d3467cd7b691e35e", GitTreeState:"clean", BuildDate:"2016-06-17T00:53:24Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
$ ./kubectl get cs
error: group map[authentication.k8s.io:0xc8203d9570 :0xc8203d9110 apps:0xc8203d9180 authorization.k8s.io:0xc8203d91f0 batch:0xc8203d92d0 componentconfig:0xc8203d9340 extensions:0xc8203d93b0 rbac.authorization.k8s.io:0xc8203d9490 federation:0xc8203d90a0 autoscaling:0xc8203d9260 policy:0xc8203d9420] is already registered```

So this appears to only report back the correct error when using exactly the same version, v1.3.0-beta.1. Are there any further debugging steps I can take to help pin this down?

**Comment by madhusudancs** [comment]
@Seth-Karlo There were a number of changes that went in between `v1.3.0-beta.1` and `v1.3.0` and I am not even sure if we tested the version skew between those two versions. Is it possible to upgrade your cluster to the latest stable 1.3.x version?

Also, what is your current-context set to (`$ kubectl config current-context`)? Is it pointing to the federation API server?

**Comment by Seth-Karlo** [comment]
@madhusudancs I can do, sure thing. It is definitely pointed at the federated API server, but I will test an upgraded version of the federated-api-server and report back.

**Comment by madhusudancs** [comment]
@Seth-Karlo Thanks! 

Also, we don't have componentstatus for federation components. So `kubectl get cs` doesn't work if you point it to a federation API server. If you want to get the componentstatuses of Kubernetes cluster components then use `kubectl --context=<kube-cluster-context>`. You can get the kube-cluster-context from `kubectl config view`

**Comment by Seth-Karlo** [comment]
@madhusudancs Yep, I know it doesn't have that option, but it should spit out the correct error message when it does not have that in my opinion, like the kubectl with v1.3.0-beta.1.

**Comment by madhusudancs** [comment]
@Seth-Karlo agreed.

**Comment by Seth-Karlo** [comment]
@nikhiljindal @madhusudancs This appears to be fixed in version 1.3.3:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:29:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.3", GitCommit:"c6411395e09da356c608896d3d9725acab821418", GitTreeState:"clean", BuildDate:"2016-07-22T20:22:25Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
$ kubectl get cs
Unable to list "componentstatuses": the server could not find the requested resource
```

**Comment by madhusudancs** [comment]
@Seth-Karlo thanks for verifying. I think this was an issue with the `v1.3.0-beta.1` release. I am closing this issue for now. If you see this issue again when you upgrade your `kubectl` or your cluster please reopen this issue or let us know.
